src/lib/services/tx/index.ts

```diff
@@ -290,24 +293,31 @@ export const useTxsByAddressLcd = (
   options: UseQueryOptions<TxsResponse> = {}
 ) => {
   const endpoint = useLcdEndpoint();
-  const { validateContractAddress } = useValidateAddress();
   const {
     chain: { bech32_prefix: prefix },
   } = useCurrentChain();
 
+  // eslint-disable-next-line sonarjs/cognitive-complexity
   const queryfn = useCallback(async () => {
     const txs = await (async () => {
-      if (search && isTxHash(search)) return getTxsByHashLcd(endpoint, search);
-
-      if (search && !validateContractAddress(search))
-        return getTxsByContractAddressLcd(
-          endpoint,
-          search as BechAddr32,
-          limit,
-          offset
+      if (search && isTxHash(search)) {
+        const txsByHash = await getTxsByHashLcd(endpoint, search);
```

Comment:
I think it might be like 
```typescript
export const useTxsByAddressLcd = (
  address: Option<BechAddr20>,
  search: Option<string>,
  limit: number,
  offset: number,
  options: UseQueryOptions<TxsResponse> = {}
) => {
  const endpoint = useLcdEndpoint();
  const {
    chain: { bech32_prefix: prefix },
  } = useCurrentChain();

  const queryfn = useCallback(async () => {
    const txs = await (async () => {
      if (!address)
        throw new Error("address is undefined (useTxsByAddressLcd)");

      if (!search || search?.trim().length === 0)
        throw new Error("search is empty (useTxsByAddressLcd)");

      if (isTxHash(search)) {
        const txsByHash = await getTxsByHashLcd(endpoint, search);

        if (txsByHash.total === 0)
          throw new Error("address is not equal to sender (getTxsByHashLcd)");

        const tx = txsByHash.items[0];
        const sender = convertAccountPubkeyToAccountAddress(
          tx.signerPubkey,
          prefix
        );

        if (address === sender) return txsByHash;

        throw new Error("address is not equal to sender (getTxsByHashLcd)");
      }

      return getTxsByAccountAddressLcd(endpoint, address, limit, offset);
    })();

    return {
      items: txs.items.map<Transaction>((tx) => ({
        ...tx,
        sender: convertAccountPubkeyToAccountAddress(tx.signerPubkey, prefix),
      })),
      total: txs.total,
    };
  }, [address, endpoint, limit, offset, prefix, search]);

  return useQuery<TxsResponse>(
    [
      CELATONE_QUERY_KEYS.TXS_BY_ADDRESS_LCD,
      endpoint,
      address,
      search,
      limit,
      offset,
    ],
    createQueryFnWithTimeout(queryfn, 20000),
    { ...options, retry: 1, refetchOnWindowFocus: false }
  );
};
```